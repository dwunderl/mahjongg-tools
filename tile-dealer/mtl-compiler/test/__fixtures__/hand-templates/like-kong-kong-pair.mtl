metadata = {
    id: "like_kong_kong_pair",
    name: "Like Kong Kong Pair",
    description: "Like kongs and Pair in different suits with matching dragons and flower pair",
    category: "ANY LIKE NUMBERS",
    catid: "1",
    image: "<black>FF <green>1111 D <red>1111 D <black>11"
}

Variations:
    suits = (b, c, d)
    
    foreach (s1 in suits):
        s2, s3 = complement(s1, suits)
        
        foreach (n in 1..9):
            pair(F)
            kong(n, s2)
            single(D, s2)
            kong(n, s3)
            single(D, s3)
            pair(n, s1)
        end
    end
