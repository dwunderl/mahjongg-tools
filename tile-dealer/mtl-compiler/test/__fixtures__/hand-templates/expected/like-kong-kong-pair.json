{
  "templates": [
    {
      "metadata": {
        "id": "1",
        "name": "Like Kong Kong Pair",
        "description": "Like kongs and Pair in different suits with matching dragons and flower pair",
        "category": "ANY LIKE NUMBERS",
        "image": "<black>FF <green>1111 D <red>1111 D <black>11"
      },
      "variations": [
        {
          "tiles": [],
          "name": "Like Kong Kong Pair",
          "description": " undefined pair"
        },
        {
          "tiles": [],
          "name": "Like Kong Kong Pair",
          "description": " s1 pair"
        }
      ]
    }
  ]
}